### ISSUE TITLE
🧪 test: BE - getFilteredCafe controller function unit tests

### ISSUE THREAD

**Issue by pablisch** [Pensano-dev/aura-mobile-app#78]
#75 should be completed BEFORE this ticket is done. By all means start this ticket but the results of tests will change as a result of ticket #75.

Not much to say here. Cover a range of test cases for valid results and bad requests.

**AC:**
Tests include:

- [ ] Valid request when given a location and required facilities.
- [ ] Valid request when given a location and wanted facilities.
- [ ] Valid request when given a location, distance and required facilities.
- [ ] Valid request when given a location, distance, wanted facilities and required facilities.
- [ ] Invalid request where location is missing.
- [ ] Invalid request where location is of wrong type.
- [ ] Invalid request where location is correct type but invalid.
- [ ] Invalid request where no facilities of any kind are provided.
- [ ] Invalid request where facilities/required facilities are of the wrong type.
- [ ] Invalid request where distance is of the wrong type.
- [ ] Invalid request where distance is invalid, e.g. too large, negative.
- [ ] All tests return valid data or a user-friendly error message and a suitable status code.
- [ ] No test crashes the server.

This is not an exhaustive list. Please add more.

**NOTE:** All invalid requests should be gracefully handled without crashing the server and return a helpful message. Please let the team know of any issue you uncover. It is not at all the job of this ticket to fix those issues but to reveal them. This is why we test.

**Comment by pablisch** [comment]
You may have noticed that the pre-requisite ticket, #75 is on hold because of possible/probable changes resulting from the use of the Google Places API. If you want to complete this ticket anyway, it may be redundant but will be full of practice and learning. Entirely up to you.
